cli/src/semgrep/commands/scan.py

```diff
@@ -454,6 +454,11 @@ def scan(
 
             version_check()
         return None
+    
+    if config and "secrets" in config:
```

Comment:
@jonas I wish there was an easy way to leverage the `engine_params` from the new `GET /api/cli/scans` endpoint here but that info is not available until we fetch the rules which happens further along when fetching the config. 

This was the easiest solution.

Reply:
Probably worth leaving this as a comment in the code for our future selves.

Reply:
Done in [f70afa6](https://github.com/semgrep/semgrep/pull/9205/commits/f70afa6daf1cfc5be3726595106e1c3ca22dd8ad)